sidechain/state/src/lib.rs

```diff
@@ -43,9 +43,9 @@ use sp_core::H256;
 use sp_std::prelude::Vec;
 use std::marker::PhantomData;
 
-/// Sidechain DB
+/// Sidechain wrapper and interface of the STF state.
```

Comment:
@clangenb will have to comment on this, maybe he has an idea what would be a better name.

Reply:
`Sidechain wrapper` is wrong in my opinion. Which sidechain does it wrap? :)

Reply:
it wraps the stf state, not the sidechain. But SidechainDB indicates a database behind it. We have a sidechain database, but on the untrusted side.

Reply:
Hmm, yes. Simply `SidechainStateApi` might be a good option.

Reply:
@mullefel told me that the more independencies the better. So I'm thinking of removing the whole StfExecution part from the "top pool executor". We have a stf executor - so why not simply use it directly:
https://github.com/integritee-network/worker/pull/520/files#diff-fd448d979e5ae8c77e49f13a55daccf783b7f2732a263b931c8d113d831e5311R60-R108

My idea would be to have "propose" choose its top pool "executor" and stf state, not the top pool executor, removing the dependency of top pool and stf.

But that's curently in progress. If you think that's not the way to go I'd be happy to talk about it :)

Reply:
Oh sorry, only just now noticed your comment was not about the executor. Sorry, ignore the comment above, it's out of place.

Reply:
haha, I wanted to ask you later, what made you write this essay. 😄